pkg/cloud/aws/services/ec2/vpc_ip_resource_limit.go

```diff
@@ -0,0 +1,459 @@
+	"i3.2xlarge":    15,
+	"i3.4xlarge":    30,
+	"i3.8xlarge":    30,
+	"i3.16xlarge":   31,
+	"i3.metal":      50,
+	"i3en.large":    10,
+	"i3en.xlarge":   15,
+	"i3en.2xlarge":  15,
+	"i3en.3xlarge":  15,
+	"i3en.6xlarge":  30,
+	"i3en.12xlarge": 30,
+	"i3en.24xlarge": 50,
+	"m1.small":      4,
+	"m1.medium":     6,
+	"m1.large":      10,
+	"m1.xlarge":     15,
+	"m2.xlarge":     15,
+	"m2.2xlarge":    30,
+	"m2.4xlarge":    30,
+	"m3.medium":     6,
+	"m3.large":      10,
+	"m3.xlarge":     15,
+	"m3.2xlarge":    30,
+	"m4.large":      10,
+	"m4.xlarge":     15,
+	"m4.2xlarge":    15,
+	"m4.4xlarge":    30,
+	"m4.10xlarge":   30,
+	"m4.16xlarge":   30,
+	"m5.large":      10,
+	"m5.xlarge":     15,
+	"m5.2xlarge":    15,
+	"m5.4xlarge":    30,
+	"m5.8xlarge":    30,
+	"m5.12xlarge":   30,
+	"m5.16xlarge":   50,
+	"m5.24xlarge":   50,
+	"m5.metal":      50,
+	"m5a.large":     10,
+	"m5a.xlarge":    15,
+	"m5a.2xlarge":   15,
+	"m5a.4xlarge":   30,
+	"m5a.8xlarge":   30,
+	"m5a.12xlarge":  30,
+	"m5a.16xlarge":  50,
+	"m5a.24xlarge":  50,
+	"m5ad.large":    10,
+	"m5ad.xlarge":   15,
+	"m5ad.2xlarge":  15,
+	"m5ad.4xlarge":  30,
+	"m5ad.12xlarge": 30,
+	"m5ad.24xlarge": 50,
+	"m5d.large":     10,
+	"m5d.xlarge":    15,
+	"m5d.2xlarge":   15,
+	"m5d.4xlarge":   30,
+	"m5d.8xlarge":   30,
+	"m5d.12xlarge":  30,
+	"m5d.16xlarge":  50,
+	"m5d.24xlarge":  50,
+	"m5d.metal":     50,
+	"p2.xlarge":     15,
+	"p2.8xlarge":    30,
+	"p2.16xlarge":   30,
+	"p3.2xlarge":    15,
+	"p3.8xlarge":    30,
+	"p3.16xlarge":   30,
+	"p3dn.24xlarge": 50,
+	"r3.large":      10,
+	"r3.xlarge":     15,
+	"r3.2xlarge":    15,
+	"r3.4xlarge":    30,
+	"r3.8xlarge":    30,
+	"r4.large":      10,
+	"r4.xlarge":     15,
+	"r4.2xlarge":    15,
+	"r4.4xlarge":    30,
+	"r4.8xlarge":    30,
+	"r4.16xlarge":   31,
+	"r5.large":      10,
+	"r5.xlarge":     15,
+	"r5.2xlarge":    15,
+	"r5.4xlarge":    30,
+	"r5.8xlarge":    30,
+	"r5.12xlarge":   30,
+	"r5.16xlarge":   50,
+	"r5.24xlarge":   50,
+	"r5.metal":      50,
+	"r5a.large":     10,
+	"r5a.xlarge":    15,
+	"r5a.2xlarge":   15,
+	"r5a.4xlarge":   30,
+	"r5a.8xlarge":   30,
+	"r5a.12xlarge":  30,
+	"r5a.16xlarge":  50,
+	"r5a.24xlarge":  50,
+	"r5ad.large":    10,
+	"r5ad.xlarge":   15,
+	"r5ad.2xlarge":  15,
+	"r5ad.4xlarge":  30,
+	"r5ad.12xlarge": 30,
+	"r5ad.24xlarge": 50,
+	"r5d.large":     10,
+	"r5d.xlarge":    15,
+	"r5d.2xlarge":   15,
+	"r5d.4xlarge":   30,
+	"r5d.8xlarge":   30,
+	"r5d.12xlarge":  30,
+	"r5d.16xlarge":  50,
+	"r5d.24xlarge":  50,
+	"r5d.metal":     50,
+	"t1.micro":      2,
+	"t2.nano":       2,
+	"t2.micro":      2,
+	"t2.small":      4,
+	"t2.medium":     6,
+	"t2.large":      12,
+	"t2.xlarge":     15,
+	"t2.2xlarge":    15,
+	"t3.nano":       2,
+	"t3.micro":      2,
+	"t3.small":      4,
+	"t3.medium":     6,
+	"t3.large":      12,
+	"t3.xlarge":     15,
+	"t3.2xlarge":    15,
+	"t3a.nano":      2,
+	"t3a.micro":     2,
+	"t3a.small":     4,
+	"t3a.medium":    6,
+	"t3a.large":     12,
```

Comment:
Can we generalize these as well? For example it looks like `*.large` = 10 with the exception of `t*.large` which is 12.

Reply:
> Can we generalize these as well? For example it looks like `*.large` = 10 with the exception of `t*.large` which is 12.

This is copied from the aws product documentation, aws did not tell us the number of elastic NICs and the linear relationship of the model in the document.